backend/src/middleware/errorHandler.ts

```
@@ -0,0 +1,259 @@
+  // Default to generic AppError
+  return toAppError(error);
+}
+
+/**
+ * Handle Prisma-specific errors
+ */
+function handlePrismaError(error: Prisma.PrismaClientKnownRequestError): AppError {
+  switch (error.code) {
+    case 'P2002':
+      // Unique constraint violation
+      const fields = error.meta?.target as string[] || [];
```

Biome (2.5.0)

[error] Other switch clauses can erroneously access this declaration.
Wrap the declaration in a block to restrict its access to the switch clause.

(lint/correctness/noSwitchDeclarations)